scripts/jointLklDM.C

```diff
@@ -167,26 +171,85 @@ void jointLklDM(TString configFileName="$GLIKESYS/rcfiles/jointLklDM.rc",Int_t s
   const Double_t* massval      = massval0+nskippedmass;
   
   // Print-out configuration info (part 2)
-  cout << "*** I/O PATH            : " << fInputDataPath << endl;
+  cout << "*** I/O PATH                 : " << fInputDataPath << endl;
   cout << "***" << endl;
-  cout << "*** LABEL               : " << label <<  endl;
-  cout << "*** PROCESS             : " << strprocess << endl;
-  cout << "*** CHANNEL             : " << channel << endl;
-  cout << "*** DATA/MC             : " << simulationlabel << endl;
+  cout << "*** LABEL                    : " << label <<  endl;
+  cout << "*** PROCESS                  : " << strprocess << endl;
+  cout << "*** CHANNEL                  : " << channel << endl;
+  Int_t nChannels = 1;
+  TString* channelval = new TString[nChannels];
+  Double_t* brval = new Double_t[nChannels];
+  if(channellist.Contains(channel))
+    {
+       channelval[0] = channel;
+       brval[0] = 1.0;
+    }
+  else
+    {
+      TObjArray *coefficients = strchannel.Tokenize("+");
+      nChannels = coefficients->GetEntries();
+      channelval = new TString[nChannels];
+      brval = new Double_t[nChannels];
+      TString factorstring, coefficientstring;
+      for (Int_t iChannel = 0; iChannel < nChannels; iChannel++) 
+        {
+          factorstring = (TString)((TObjString *)(coefficients->At(iChannel)))->String();
+          TObjArray *factors = factorstring.Tokenize("*");
+          if (factors->GetEntries() != 2)
+            {
+              cout << " ## Oops! Something went wrong with the parsing of the channel (" << channel << ")!  <---------------- FATAL ERROR!!!"<< endl;
+              return;
+            }
+          for (Int_t i = 0; i < factors->GetEntries(); i++) 
+            {
+              coefficientstring = (TString)((TObjString *)(factors->At(i)))->String();
+              if (coefficientstring.IsFloat() && i == 0)
+                brval[iChannel] = coefficientstring.Atof();
+              else if(channellist.Contains(coefficientstring) && i == 1)
+                channelval[iChannel] = coefficientstring;
+              else
+                {
+                  cout << " ## Oops! Something went wrong with the parsing of the channel (" << channel << ")!  <---------------- FATAL ERROR!!!"<< endl;
+                  return;
+                }
+              TString coefficientstring = (TString)((TObjString *)(factors->At(i)))->String();
+            }
+        }
+      // how many and which channel values are we considering, if a custom linear combination of channels is selected?
+      cout << " ** Custom linear combination:" << endl;
+      cout << "  * Number of coefficients   : " << nChannels << endl;
+      cout << "  * Channel values           : ";
```

Comment:
These three lines should be replaced by one, just reporting the channel as a string of the type "br1*channel1+...+brn*channeln"
It is important to not just return channel, but to write it again with the values of nChannels, brval and channelval, previously decoded, in this way you can spot mistakes in your decoding function

Reply:
True!